### PR TITLE
Geometry cant templates

### LINKED ISSUE #221: Alignment Geometry Cant Template
## Description & Tasks
Geometry template for the definition Cant within Alignment As Sibling to Alignment Geometry Gradient. this is the geometry component of the related semantic template Issue #228 Alignment Cant and #256

- [x] Author Alignment Geometry Cant Template
- [x] Documentation of Alignment Geometry Cant Template

### Discussion Points
TBD

### Links
[Miro Element](https://miro.com/app/board/o9J_l5rJCLk=/?moveToWidget=3074457362581109449&cot=14)
[Miro Task Card](https://miro.com/app/board/o9J_l5rJCLk=/?moveToWidget=3074457362911087566&cot=14)

### Decisions 
1. Relating to #229 and #233 cant shall be used in certain geometry intensive cases such as sleeper, catenary post placement and geometric representations, structural gauge etc. The relative positioning in the semantic layer shall be covered with #233.

### PR DESCRIPTION
**Pull request by SergejMuhic**
Resolves #221 

Actions done:
- Gradient curve partial template
- use of the partial in Alignment Gradient curve geometry and Alignment cant geometry

### CODE REVIEW

**Comment by AlexBrad1eyCT** [comment]
@czapplitec @apinzenoehler please review so this can be merged we are on a deadline!!

**Review by AlexBrad1eyCT — APPROVED** [review]
LFTM we can address any needed Identification during the GU for Alignment

**Comment by czapplitec** [comment]
I agree to approve it firstly, but I have one concern for Alignment Geometry Gradient that can addressed in GU and MVD phase:
In my understanding, in the cant scenario, IfcGradientCurve should still be the representation of IfcAlignment (same with Alignment Geometry Gradient), because IfcGradientCurve is used for positioning. IfcSegmentedReferenceCurve can be the representation of IfcAlignmentCant. But I feel it is difficult to describe the entire structure with current mvdXML: in the CT, we cannot say that the IfcAlignmentCant.Representation.Representations.Items/IfcSegmentedReferenceCurve.BaseCurve is IfcAlignment.Representation.Representations.Items/IfcGradientCurve.

**Review by czapplitec — APPROVED** [review]

**Comment by AlexBrad1eyCT** [comment]
Thank you @czapplitec for the approval, I think we should be able to cover this in the GU & Alignment MVD with the use of new features in mvdXML 1.2 I will be merging this Pull request as to including it in the end of October deliverable

File: IFC4x3/Templates/Partial Templates/Geometry/Curve Axis Geometry/Gradient Curve/DocTemplateDefinition.xml [hunk 1]
```diff
@@ -1,9 +1,41 @@
 <?xml version="1.0" encoding="utf-8"?>
-<DocTemplateDefinition xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Gradient Curve" UniqueId="cf0d4709-dc5b-484e-ae24-946553d6a4e1" Type="IfcGradientCurve">
+<DocTemplateDefinition xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="Gradient_Curve_3F3KS9t5j8Jguab6LJrgJX" Name="Gradient Curve" UniqueId="cf0d4709-dc5b-484e-ae24-946553d6a4e1" Type="IfcGradientCurve">
 	<Rules>
 		<DocModelRuleAttribute Name="BaseCurve">
```
[the remark this answers is not in the document]
Comment (SergejMuhic) [reply IFC4x3/Templates/Partial Templates/Geometry/Curve Axis Geometry/Gradient Curve/DocTemplateDefinition.xml]: Struggling to see the purpose in this. It is restricted to `IfcCompositeCurve` in the template.

**Comment by SergejMuhic** [comment]
@AlexBrad1eyCT Please read my comment. The ruleIDs need some work. Representation identifiers have not been set. Since when do we not have a quality gate?

**Pull request merged**